dashboard/app/models/user.rb

```diff
@@ -2261,6 +2259,40 @@ def providers
     end
   end
 
+  def num_failed_section_attempts
+    properties['section_attempts'] || 0
+  end
+
+  def reset_section_attempts?
+    num_failed_section_attempts == 0 || (DateTime.now - DateTime.parse(properties['section_attempts_last_reset'])).to_i > 0
```

Comment:
Why do we want to return true if num_failed_section_attempts == 0?

Reply:
My code in the method `num_failed_section_attempts` originally returned 0 if properties['section_attempts'] was not initialized. However, I've changed it to return nil in this case and adjusted `reset_section_attempts?` accordingly; hopefully that improves clarity.